src/components/ResultFavoritesButton/ResultFavoritesButton.tsx

```diff
@@ -8,14 +8,22 @@ interface ResultFavoritesButtonProps {
 }
 
 export default function ResultFavoritesButton(props: ResultFavoritesButtonProps) {
+  const [isFavorited, setIsFavorited] = useState(false);
   const { item, price } = props;
   const { getAddToFavoritesButtonProps } = useContext(QuizContext);
 
+  const clickHandler = () => {
```

Comment:
Lets not have any logic in presentational components and have the logic in hooks